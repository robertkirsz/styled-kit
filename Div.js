module.exports=function(e){var t={};function n(r){if(t[r])return t[r].exports;var i=t[r]={i:r,l:!1,exports:{}};return e[r].call(i.exports,i,i.exports,n),i.l=!0,i.exports}return n.m=e,n.c=t,n.d=function(e,t,r){n.o(e,t)||Object.defineProperty(e,t,{enumerable:!0,get:r})},n.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},n.t=function(e,t){if(1&t&&(e=n(e)),8&t)return e;if(4&t&&"object"==typeof e&&e&&e.__esModule)return e;var r=Object.create(null);if(n.r(r),Object.defineProperty(r,"default",{enumerable:!0,value:e}),2&t&&"string"!=typeof e)for(var i in e)n.d(r,i,function(t){return e[t]}.bind(null,i));return r},n.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return n.d(t,"a",t),t},n.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},n.p="",n(n.s=6)}([function(e,t,n){"use strict";e.exports=n(4)},function(e,t){e.exports=function(e){return e.webpackPolyfill||(e.deprecate=function(){},e.paths=[],e.children||(e.children=[]),Object.defineProperty(e,"loaded",{enumerable:!0,get:function(){return e.l}}),Object.defineProperty(e,"id",{enumerable:!0,get:function(){return e.i}}),e.webpackPolyfill=1),e}},function(e,t){e.exports=require("react")},function(e,t){e.exports=require("styled-components")},function(e,t,n){"use strict";Object.defineProperty(t,"__esModule",{value:!0});var r=function(e){return e&&"object"==typeof e&&"default"in e?e.default:e}(n(2)),i=function(e,t){if(!(e instanceof t))throw new TypeError("Cannot call a class as a function")},o=function(e,t){if(!e)throw new ReferenceError("this hasn't been initialised - super() hasn't been called");return!t||"object"!=typeof t&&"function"!=typeof t?e:t},s=function(e){function t(){return i(this,t),o(this,e.apply(this,arguments))}return function(e,t){if("function"!=typeof t&&null!==t)throw new TypeError("Super expression must either be null or a function, not "+typeof t);e.prototype=Object.create(t&&t.prototype,{constructor:{value:e,enumerable:!1,writable:!0,configurable:!0}}),t&&(Object.setPrototypeOf?Object.setPrototypeOf(e,t):e.__proto__=t)}(t,e),t.prototype.render=function(){return r.Children.only(this.props.children)},t}(r.Component);t.AppContainer=s,t.hot=function(){return function(e){return e}},t.areComponentsEqual=function(e,t){return e===t},t.setConfig=function(){},t.cold=function(e){return e}},,function(e,t,n){"use strict";(function(e){Object.defineProperty(t,"__esModule",{value:!0});var r=n(3),i=function(e){return e&&e.__esModule?e:{default:e}}(r),o=n(7);!function(){var t=n(0).enterModule;t&&t(e)}();var s="position: absolute; top: 0; right: 0; bottom: 0; left: 0;",u=i.default.div.withConfig({displayName:"Div___default"})(["display:",";"," "," "," "," "," "," "," "," "," "," "," "," "," "," "," "," "," "," "," "," "," "," "," "," "," "," "," "," ",""],function(e){return e.inline?e.block?"inline-block":e.grid?"inline-grid":!0===e.flex?"inline-flex":"inline":!0===e.flex?"flex":e.block?"block":e.grid?"grid":"flex"},function(e){if(!e.block||!e.inline||e.flex){var t="";return e.row&&(t+="flex-direction: row;"),e.rowReverse&&(t+="flex-direction: row-reverse;"),e.column&&(t+="flex-direction: column;"),e.columnReverse&&(t+="flex-direction: column-reverse;"),e.justifyContent&&(t+="justify-content: "+e.justifyContent+";"),e.justifyStart&&(t+="justify-content: flex-start;"),e.justifyEnd&&(t+="justify-content: flex-end;"),e.justifyCenter&&(t+="justify-content: center;"),e.justifyBetween&&(t+="justify-content: space-between;"),e.justifyAround&&(t+="justify-content: space-around;"),e.justifyEvenly&&(t+="justify-content: space-evenly;"),e.alignItems&&(t+="align-items: "+e.alignItems+";"),e.itemsStart&&(t+="align-items: flex-start;"),e.itemsEnd&&(t+="align-items: flex-end;"),e.itemsCenter&&(t+="align-items: center;"),e.itemsBaseline&&(t+="align-items: baseline;"),e.itemsStretch&&(t+="align-items: stretch;"),e.alignContent&&(t+="align-content: "+e.alignContent+";"),e.contentStart&&(t+="align-content: flex-start;"),e.contentEnd&&(t+="align-content: flex-end;"),e.contentCenter&&(t+="align-content: center;"),e.contentBetween&&(t+="align-content: space-between;"),e.contentArouns&&(t+="align-content: space-around;"),e.contentStretch&&(t+="align-content: stretch;"),e.alignSelf&&(t+="align-self: "+e.alignSelf+";"),e.selfAuto&&(t+="align-self: auto;"),e.selfStart&&(t+="align-self: flex-start;"),e.selfEnd&&(t+="align-self: flex-end;"),e.selfCenter&&(t+="align-self: center;"),e.selfBaseline&&(t+="align-self: baseline;"),e.selfStretch&&(t+="align-self: stretch;"),t}},function(e){return e.wraps&&"flex-wrap: wrap;"},function(e){var t=e.area;return"string"==typeof t&&"grid-area: "+t+";"},function(e){var t=e.order;return(0,o.isAlphaNumeric)(t)&&"order: "+t+";"},function(e){var t=e.flexNone,n=e.flex;return t?"flex: none;":(0,o.isAlphaNumeric)(n)?(0,r.css)(["flex:",";"],n):void 0},function(e){var t=e.absolute,n=e.fixed,r=e.relative;return t?"position: absolute;":n?"position: fixed;":r?"position: relative;":void 0},o.createPosition,function(e){var t=e.width;return t&&(0,r.css)(["width:",";"],(0,o.withUnit)(t))},function(e){var t=e.height;return t&&(0,r.css)(["height:",";"],(0,o.withUnit)(t))},function(e){var t=e.minWidth;return t&&(0,r.css)(["min-width:",";"],(0,o.withUnit)(t))},function(e){var t=e.minHeight;return t&&(0,r.css)(["min-height:",";"],(0,o.withUnit)(t))},function(e){var t=e.maxWidth;return t&&(0,r.css)(["max-width:",";"],(0,o.withUnit)(t))},function(e){var t=e.maxHeight;return t&&(0,r.css)(["max-height:",";"],(0,o.withUnit)(t))},function(e){return e.fullHeight&&(0,r.css)(["min-height:100vh;"])},function(e){var t=e.margin,n=e.m;return(t||n)&&(0,r.css)(["margin:",";"],(0,o.withUnit)(t||n))},(0,o.createSpaces)("margin"),function(e){var t=e.padding,n=e.p;return(t||n)&&(0,r.css)(["padding:",";"],(0,o.withUnit)(t||n))},(0,o.createSpaces)("padding"),function(e){var t=e.z;return(0,o.isAlphaNumeric)(t)&&(0,r.css)(["z-index:",";"],t)},o.createLists,function(e){return e.layer&&s},function(e){var t=e.square;return t&&(0,r.css)(["width:",";height:",";"],(0,o.withUnit)(t),(0,o.withUnit)(t))},function(e){return e.clickable&&"cursor: pointer;"},function(e){return e.noPointerEvents&&(0,r.css)(["pointer-events:none;"])},function(e){var t=e.background;return t&&(0,r.css)(["background:",";"],t)},function(e){var t=e.backgroundImage;return t&&(0,r.css)(["background-image:url(",");"],t)},function(e){var t=e.cover,n=e.contain;return(t||n)&&(0,r.css)(["background-size:",";"],t?"cover":"contain")},function(e){return e.overlay&&(0,r.css)(["position:",';&::after{content:"";'," background:",";opacity:0.2;pointer-events:none;}"],e.absolute?"absolute":"relative",s,e.overlay)},function(e){return void 0!==e.visible&&(0,r.css)(["transition:opacity 0.3s;",""],!1===e.visible&&"opacity: 0;")});t.default=u,function(){var t=n(0).default,r=n(0).leaveModule;t&&(t.register(s,"layerStyles","/Users/robert/dev/styled-kit/src/components/Div.js"),t.register(u,"default","/Users/robert/dev/styled-kit/src/components/Div.js"),r(e))}()}).call(this,n(1)(e))},function(e,t,n){"use strict";(function(e){Object.defineProperty(t,"__esModule",{value:!0}),function(){var t=n(0).enterModule;t&&t(e)}();var r=t.isNumber=function(e){return"number"==typeof e},i=t.isString=function(e){return"string"==typeof e},o=t.isAlphaNumeric=function(e){return i(e)||r(e)},s=t.withUnit=function(e){return r(e)?e+"px":e},u=["Left","Right","Top","Bottom"],c=t.createSpaces=function(e){return function(t){var n="";return u.forEach(function(r){var i=t[""+e[0]+r];void 0!==i&&(n+=e+"-"+r.toLowerCase()+": "+s(i)+";")}),n}},a=t.createPosition=function(e){var t="";return u.forEach(function(n){var r=e[""+n.toLowerCase()];void 0!==r&&(t+=n.toLowerCase()+": "+s(r)+";")}),t},l=t.createLists=function(e){var t="";return u.forEach(function(n,r){var i=e["list"+n];if(void 0!==i){var o=!0===i?"8px":s(i);t+="> *:not(:"+(r%2?"last":"first")+"-child) { margin-"+n.toLowerCase()+": "+o+"; }"}}),t};!function(){var t=n(0).default,f=n(0).leaveModule;t&&(t.register(r,"isNumber","/Users/robert/dev/styled-kit/src/utils/index.js"),t.register(i,"isString","/Users/robert/dev/styled-kit/src/utils/index.js"),t.register(o,"isAlphaNumeric","/Users/robert/dev/styled-kit/src/utils/index.js"),t.register(s,"withUnit","/Users/robert/dev/styled-kit/src/utils/index.js"),t.register(u,"directions","/Users/robert/dev/styled-kit/src/utils/index.js"),t.register(c,"createSpaces","/Users/robert/dev/styled-kit/src/utils/index.js"),t.register(a,"createPosition","/Users/robert/dev/styled-kit/src/utils/index.js"),t.register(l,"createLists","/Users/robert/dev/styled-kit/src/utils/index.js"),f(e))}()}).call(this,n(1)(e))}]);