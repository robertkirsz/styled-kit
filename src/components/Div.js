import styled from 'styled-components'

import stuff from 'stuff'
import { camelToKebab } from 'utils'

function doStuff(props) {
  const mgKeys = Object.keys(props.theme.styledKitMediaQueries || {})

  const out = Object.keys(props).reduce((css, prop) => {
    const value = stuff[prop]

    if (typeof value !== 'function') return css
    if (mgKeys.includes(prop)) return css
    if (['', false, undefined].includes(value(props[prop], props))) return css

    // if (props.debug) console.log({ prop, value: value(props[prop], props) })

    return `${css}${value(props[prop], props)}`
  }, '')

  // if (props.debug) console.log(out)
  return out
}

function doMediaQueriesStuff(props) {
  if (!props.theme || !props.theme.styledKitMediaQueries) return

  const queryNames = Object.keys(props.theme.styledKitMediaQueries)

  if (!queryNames.length) return

  const queryNameToValuesMap = queryNames.reduce((all, query) => {
    const declaration = props[query]

    if (!declaration) return all

    const allDeclarations =
      typeof declaration === 'string'
        ? declaration
        : Object.keys(props[query]).reduce((all, property) => {
            const value = props[query][property]
            let foo = stuff[property]

            if (!foo) return `${all}${camelToKebab(property)}:${value};`

            const declaration = typeof foo === 'function' ? foo(value, props) : foo

            return `${all}${declaration}`
          }, '')

    return { ...all, [query]: allDeclarations }
  }, {})

  const cssString = queryNames.reduce((all, query) => {
    if (!queryNameToValuesMap[query]) return all

    return `${all}@media ${props.theme.styledKitMediaQueries[query]} {${queryNameToValuesMap[query]}}`
  }, '')

  return cssString
}

export default styled.div`
  display: ${({ inline }) => (inline ? 'inline-flex' : 'flex')};
  ${doStuff}
  ${doMediaQueriesStuff}
`
