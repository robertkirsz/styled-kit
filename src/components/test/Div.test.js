import React from 'react'
import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'

import Div, { stuff } from '../Div'

describe('<Div />', () => {
  const handleDone = jest.fn()

  it('Matches snapshot', () => {
    const props = Object.keys(stuff).reduce((result, prop) => {
      if (!stuff[prop]) return result
      try {
        return { ...result, [prop]: stuff[prop](1) }
      } catch {
        return { ...result, [prop]: stuff[prop] }
      }
    }, {})

    render(
      <ThemeProvider theme={{}}>
        <Div data-testid="test" {...props} onDone={handleDone} />
      </ThemeProvider>
    )
    console.log(handleDone.mock.calls[0][0].replace(/;/g, ';\n'))
    expect(handleDone.mock.calls[0][0]).toMatchSnapshot()
  })
})
