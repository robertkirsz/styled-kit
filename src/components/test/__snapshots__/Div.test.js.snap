// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Div /> Matches snapshot 1`] = `"position: absolute;position: fixed;position: relative;flex: none;flex-direction: row;flex-direction: row-reverse;flex-direction: column;flex-direction: column-reverse;flex-wrap: wrap;justify-content: flex-start;justify-content: flex-end;justify-content: center;justify-content: space-between;justify-content: space-around;justify-content: space-evenly;align-items: flex-start;align-items: flex-end;align-items: center;align-items: baseline;align-items: stretch;align-content: flex-start;align-content: flex-end;align-content: center;align-content: space-between;align-content: space-around;align-content: stretch;align-self: auto;align-self: flex-start;align-self: flex-end;align-self: center;align-self: baseline;align-self: stretch;position: absolute; top: 0; right: 0; bottom: 0; left: 0;display: none;top: top: 1px;;right: right: 1px;;bottom: bottom: 1px;;left: left: 1px;;margin: margin: 1px;;margin-top: margin-top: 1px;;margin-right: margin-right: 1px;;margin-bottom: margin-bottom: 1px;;margin-left: margin-left: 1px;;padding: padding: 1px;;padding-top: padding-top: 1px;;padding-right: padding-right: 1px;;padding-bottom: padding-bottom: 1px;;padding-left: padding-left: 1px;;> *:not(:first-child) { margin-left: > *:not(:first-child) { margin-left: 1px; }; }> *:not(:last-child) { margin-right: > *:not(:last-child) { margin-right: 1px; }; }> *:not(:first-child) { margin-top: > *:not(:first-child) { margin-top: 1px; }; }> *:not(:last-child) { margin-bottom: > *:not(:last-child) { margin-bottom: 1px; }; }cursor: pointer;pointer-events: none;"`;
